src/Sylius/Bundle/UserBundle/Provider/AbstractUserProvider.php

```diff
@@ -38,7 +38,7 @@ public function loadUserByUsername($username): UserInterface
         $user = $this->findUser($username);
 
         if (null === $user) {
-            throw new UsernameNotFoundException(
+            throw new UserNotFoundException(
```

Comment:
@Zales0123 This may be a bc-break, but we can't create a polyfill cause it would not work on try catch.

Reply:
It shouldn't be considered a BC break IMO, given that any concerned `catch` block would emit a proper deprecation already (https://github.com/symfony/symfony/blob/5.4/src/Symfony/Component/Security/Core/Exception/UsernameNotFoundException.php#L14).

Reply:
However, given the current approach (https://github.com/Sylius/Sylius/pull/13969#issuecomment-1159379884), a way to let ppl opt-in into the new behavior could be to add a constructor argument in this class to be defined via bundle config in order to throw the expected exception.
This deprecation layer (arg + config) could then be deprecated in the next major of Sylius when 5.4 would be dropped.

Reply:
we can also add our custom exception, which will implement `UsernameNotFoundException` if exists and `UsernameNotFoundException` otherwise. This way, the default behaviour will stay the same and Sf6 will already cover requirement of having this class.

Reply:
Indeed, I remember having this idea, but I wanted yours 👍

Reply:
>which will implement `UsernameNotFoundException` if exists and `UsernameNotFoundException` otherwise

Both classes exist in 5.4, so we cannot rely on a class existence check, since we cannot detect which one is used in the `catch` in user-land code, this needs to be controlled and opt-in by the end-user.

Reply:
but we are maintaining BC - if you had the code that was working up to this moment, nothing changes. We do not trigger new exception tho. With the current implementation, you will be always able to migrate to polifiled class (however it should be discouraged). Once moved to Sf6, the previous exception won't be existing, so you had to changed your code already. Am I missing something?

Also, don't get me wrong, I don't want to insist too much. I'm just looking for flaws in what we have now :)